### ISSUE TITLE
[Feature Request] Custom single copy marker gene sets

### ISSUE THREAD

**Issue by jolespin** [cmks/DAS_Tool#69]
I want to use `DASTool` for Eukaryotic and Candidate phyla radiation bin refinement.  I planned on using the [Protista_83](https://github.com/merenlab/anvio/tree/master/anvio/data/hmm/Protista_83) and [CPR_43](https://github.com/Ecogenomics/CheckM/tree/master/custom_marker_sets) marker sets.

Is this possible already? If not, can this be a feature in the future?

Would this be difficult?

**Comment by jolespin** [comment]
If this isn't in the cards, do you have any suggestions on how to hack the code so this will work?

**Comment by cmks** [comment]
Hi @jolespin,
After doing some major code refactoring, I’ve now actually been starting to build a prototype which can handle custom marker gene sets. It still needs some more work but I think I’ll have a version ready which you can take for a spin by next week or so.

**Comment by jolespin** [comment]
That's awesome!  More than willing to help test this out if you need some outside prototyping.

**Comment by cmks** [comment]
If you check out the [dev_customSCG](https://github.com/cmks/DAS_Tool/tree/dev_customSCG) branch, you can try running DAS Tool using your own single copy gene sets. I've included a description and some examples in the [README:customSCG](https://github.com/cmks/DAS_Tool/tree/dev_customSCG#using-custom-scg-libraries). Note that the command line and dependencies (HMMER3 and some R packages; see [README:dependencies](https://github.com/cmks/DAS_Tool/tree/dev_customSCG#dependencies)) have changed a bit. As of now, the file formats of the SCG sets in the [ANVIO repo](https://github.com/merenlab/anvio/tree/master/anvio/data/hmm) are supported. However, I have not tested how well the bin selection performs using these sets, so you may want to verify the results for yourself.

Also, it is probably not a good idea to use a CPR specific marker gene set set as these genes should be already covered by the default set of bacterial SCG. Using them without or even in combination with the bacterial set could lead to a completeness overestimation of non-CPR species.

In any case, I'm curious to hear how well the protist-bin selection works! Also, let me know if you find any bugs.

**Comment by jolespin** [comment]
I'm going to try it out tomorrow morning and getting it installed right now. I made a conda environment w/ DAS_Tool and just downloaded the dev version.  Which files do I need to replace to install in my environment from the repository?

**Comment by cmks** [comment]
You need to replace `DAS_Tool` in the main dir all files in `src`. Also make sure to install the required R packages: 
```
R -e "install.packages(c('data.table','magrittr','docopt','rhmmer'), repos='http://cran.us.r-project.org')"
```
The DAS_Tool R-package is not needed anymore.

The above error indicates an issue with the command line e.g. a misspelling of the used options. Unfortunately, there is a bug in the current version of the command line parser R-package which is not returning any information about what the problem is.... However, your command looks good to me. Can you check if you're actually using the  [dev_customSCG](https://github.com/cmks/DAS_Tool/tree/dev_customSCG) branch and not the [dev](https://github.com/cmks/DAS_Tool/tree/dev) branch? The latter does not have the options `--customDbDir` and `--useCustomDbOnly`.

**Comment by jolespin** [comment]
Is this production ready?

**Comment by cmks** [comment]
Not quite. There are still a few bugs to be fixed and tests to be run. Schedule is to merge by the end of Feb.

**Comment by AmaliT** [comment]
Hi, Is there an update on this?
